### PR TITLE
#432 Adds support for Java 14 records

### PR DESCRIPTION
**Pull request by fdutton**

### CODE REVIEW

**Comment by fdutton** [comment]
Is there anything I can do to get this pull-request merged?

**Comment by jmisur** [comment]
Thank you for the submition. It seems quite significant but we'll try to review in due time.

File: spring-auto-restdocs-json-doclet-jdk9/src/test/java/capital/scalable/restdocs/jsondoclet/ExtractDocumentationAsJsonDocletTest.java [hunk 1]
```diff
@@ -19,32 +19,89 @@
...
     public void testDocumentedClass() throws IOException, JSONException {
-        String generated = IOUtils.toString(
-                new FileInputStream(new File("target/generated-javadoc-json/" + JSON_PATH)), UTF_8);
-        String expected = IOUtils.toString(
-                this.getClass().getClassLoader().getResourceAsStream(JSON_PATH), UTF_8);
-        JSONAssert.assertEquals(expected, generated, false);
+        Path source = SRC_PATH.resolve("capital/scalable/restdocs/jsondoclet/DocumentedClass.java");
+        Iterable<JavaFileObject> compilationUnits = compilationUnits(source);
+
+        DocumentationTask task = ToolProvider.getSystemDocumentationTool().getTask(null, null, null, ExtractDocumentationAsJsonDoclet.class, args, compilationUnits);
```
Comment (jmisur) [inline spring-auto-restdocs-json-doclet-jdk9/src/test/java/capital/scalable/restdocs/jsondoclet/ExtractDocumentationAsJsonDocletTest.java:69]: nice!

**Review by jmisur — COMMENTED** [review]
Great looking stuff. Can you please merge from master and check again if the Jackson version really needs to be overwritten and if the backwards incompatible changes were not solved already. Maybe they even upgraded this in the newer Spring-Boot. It would be preferred if it comes from the spring pom.

File: spring-auto-restdocs-json-doclet-jdk9/src/test/java/capital/scalable/restdocs/jsondoclet/ExtractDocumentationAsJsonDocletTest.java (hunk 1 above)
Comment (fdutton) [reply spring-auto-restdocs-json-doclet-jdk9/src/test/java/capital/scalable/restdocs/jsondoclet/ExtractDocumentationAsJsonDocletTest.java:69]: The spring-auto-restdocs parent POM specifies version 2.11.2 of Jackson Databind. It does not come from Spring. However, I have updated the POM to allow any version after 2.11.1 with the exception of 2.12.0 and added a comment stating that the version specification may be remove once the parent POM specifies a version after 2.12.0.

Comment (jmisur) [reply spring-auto-restdocs-json-doclet-jdk9/src/test/java/capital/scalable/restdocs/jsondoclet/ExtractDocumentationAsJsonDocletTest.java:69]: True, the doclet is completely independent of spring 🤦

Comment (fdutton) [reply spring-auto-restdocs-json-doclet-jdk9/src/test/java/capital/scalable/restdocs/jsondoclet/ExtractDocumentationAsJsonDocletTest.java:69]: @jmisur What are the next steps to get this merged into your master branch? FWIW, I've been using a custom build that includes this code in my company's build pipeline for almost four months without issue.

Comment (jmisur) [reply spring-auto-restdocs-json-doclet-jdk9/src/test/java/capital/scalable/restdocs/jsondoclet/ExtractDocumentationAsJsonDocletTest.java:69]: That's great to hear. I wanted to reply yesterday about possibility to put some example into samples, but let's not overdo it too much. I'll merge this and do a release.

**Review by jmisur — APPROVED** [review]

**Review by fbenz — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**